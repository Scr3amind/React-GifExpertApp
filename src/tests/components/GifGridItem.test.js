import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente <GifGridItem />', () => {

    const testTitle = 'titulo';
    const testUrl = 'https://demo.jpg';
    const wrapper = shallow(<GifGridItem title={testTitle} url={testUrl} />);

    test('debe mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    })
    
    test('debe de tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(testTitle);
        
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        
        expect(img.prop('src')).toBe(testUrl);
        expect(img.prop('alt')).toBe(testTitle);
    })

    test('debe de tener animate___fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect(className.includes('animate__fadeIn')).toBe(true);
        
    })
    
    
    
    
})
