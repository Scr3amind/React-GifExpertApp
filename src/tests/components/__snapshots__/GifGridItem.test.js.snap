// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://demo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
